### ISSUE TITLE
feature request: allow selection of step range in histogram visualization

### ISSUE THREAD

**Issue by pluskid** [tensorflow/tensorboard#2041]
Currently the histogram have a default range and sampling mechanism for the training steps axis. It would be great if some mechanism is added to allow choosing the range of steps to visualize, just like what we can do in scalar plots. 

The rationale is that it is quite common to continue training for a while after convergence. In this case, the default range and sampling mechanism will mostly show histograms *after* convergence. On the other hand, the range from the beginning to convergence might be more interesting to the researchers to investigate. Allowing the users to specify the range will make the tool a lot more useful.

Alternatively, it could at least allow the users to export the data points (just like the scalar plots) so that the users can do their own visualization to choose the range.

**Comment by shashvatshahi1998** [comment]
@stephanwlee can you please guide me regarding this issue, I want to contribute??

**Comment by stephanwlee** [comment]
Hey @shashvatshahi1998, thanks for wanting to contribute. We did explore this with https://github.com/tensorflow/tensorboard/pull/1588 but wanted to take more generic by allowing the same on x-axis to resolve the issue like #405.

Before you improve the closed pull request, please do note that there are multiple ways of "zooming" in x-axis: (1) you may just change the domain in the chart or (2) you may re-bin based on new domain. Both imo are valid and have drawbacks so you may want to think about them and perhaps gather more requirements from the other issues.

**Comment by shashvatshahi1998** [comment]
@stephanwlee  I inspected both issues link which you mentioned , they are related to zooming issue in tensorboard and this issue is regarding selection of step range how these two can be related, please provide some insights.

**Comment by stephanwlee** [comment]
I may have read too much in between lines. 

@pluskid When you said scalar plots have mechanism to choose range of steps  to visualize, what do you specifically mean? Also, can you kindly tell us what sampling are you referring to? If you are referring to the reservoir sampling, it is done on all steps, not within the range of steps you choose. Thanks!

**Comment by johnnytshi** [comment]
In the distrubtion tab, in my experience, if I train for 15k steps, histogram onlys shows the last 5k (judging from the x-axis)

Does that mean the data displayed is only for the last 5k steps?
Or the histogram shows everything, but sampled?

Related question: what is reservoir sampling?

The main issue is I completely miss the histogram before and during convergence
